### PR TITLE
Adding the NOTEXISTS operator

### PR DESCRIPTION
**Pull request by jackchoi**

### CODE REVIEW

File: spec/lib/atlas/lexer_spec.rb [hunk 1]
```diff
@@ -121,27 +121,35 @@
...
+
+        it do
+          is_expected.to match_atlas_tokens(Atlas::ValueToken.new('shipping_address'), Atlas::NotExistsToken)
+        end
+      end
     end
 
     context 'operator composition' do
       let(:input) do
-        %(sparkAssigned NOTEQ 2018-12-01 AND
```
Comment (ascoppa) [inline spec/lib/atlas/lexer_spec.rb:128]: This seems a bit unrelated to the PR main goal ... is the intent to make the spec more generic?

Comment (jackchoi) [reply spec/lib/atlas/lexer_spec.rb:128]: Yes, I'm not sure if we even need this context tbh

**Review by ascoppa — APPROVED** [review]
